lib/mayaUsd/fileio/jobs/writeJob.cpp

```diff
@@ -112,6 +112,112 @@ static TfToken _GetFallbackExtension(const TfToken& compatibilityMode)
     return UsdMayaTranslatorTokens->UsdFileExtensionDefault;
 }
 
+class AutoUpAxisChanger
```

Comment:
This is how change of axis during export is done: we temporarily rotate the Maya scene so that we can rely on the existing correct behavior already written in Maya to handle all possible nodes properly.